### PR TITLE
Add filter version of plugins

### PR DESCRIPTION
**Pull request by cosmo0920**
* add filter version of parser plugin
* add filter version of deparser plugin

These plugins are simply converted from output plugins.

### CODE REVIEW

File: lib/fluent/plugin/filter_deparser.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
+class Fluent::DeparserFilter < Fluent::Output
+  Fluent::Plugin.register_filter('deparser', self)
+
+  config_param :tag, :string, :default => nil
+  config_param :format, :string
+  config_param :format_key_names, :string
+  config_param :key_name, :string, :default => 'message'
+  config_param :reserve_data, :bool, :default => false
+
+  def april_fool_filter_stream(tag, es)
```
Comment (cosmo0920) [inline lib/fluent/plugin/filter_deparser.rb]: Should be keep an Easter egg?

Comment (tagomoris) [inline lib/fluent/plugin/filter_deparser.rb]: No :D

Comment (cosmo0920) [inline lib/fluent/plugin/filter_deparser.rb]: Oh, that's too bad.... :-/

**Comment by repeatedly** [comment]
@tagomoris How about this?

**Comment by tagomoris** [comment]
Sorry for late reply. I'll review right now.

File: lib/fluent/plugin/filter_deparser.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+class Fluent::DeparserFilter < Fluent::Output
+  Fluent::Plugin.register_filter('deparser', self)
+
+  config_param :format, :string
+  config_param :format_key_names, :string
+  config_param :key_name, :string, :default => 'message'
```
Comment (tagomoris) [inline lib/fluent/plugin/filter_deparser.rb]: use `default: 'message'` style hash literal for new code

File: lib/fluent/plugin/filter_deparser.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    rescue ArgumentError
+      raise Fluent::ConfigError, "mismatch between placeholder of format and format_key_names"
+    end
+  end
+
+  def filter_stream(tag, es)
+    new_es = Fluent::MultiEventStream.new
+    if @reserve_data
+      es.each {|time,record|
+        record.update({@key_name => (@format % @format_key_names.map{|k| record[k]})})
```
Comment (tagomoris) [inline lib/fluent/plugin/filter_deparser.rb]: Do not `update` original record. It breaks records which are read from other plugins.
Use `new_es.add(time, record.merge({...})`.

File: lib/fluent/plugin/filter_deparser.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        new_es.add(time, record)
+      }
+    else
+      es.each {|time,record|
+        new_record = {@key_name => (@format % @format_key_names.map{|k| record[k]})}
```
Comment (tagomoris) [inline lib/fluent/plugin/filter_deparser.rb]: ditto.

**Comment by tagomoris** [comment]
Added some comment

**Comment by repeatedly** [comment]
@tagomoris if you okay, please merge this PR and release new version.

**Comment by tagomoris** [comment]
LGTM. Thank you for contribution!

**Comment by tagomoris** [comment]
`v0.6.0` released!

**Comment by cosmo0920** [comment]
:+1:

**Comment by repeatedly** [comment]
We need filter section on README for newbies.

**Pull request merged**